src/pydata_sphinx_theme/__init__.py

```diff
@@ -951,17 +951,16 @@ def _overwrite_pygments_css(app, exception=None):
         # see if user specified a light/dark pygments theme, if not, use the
         # one we set in theme.conf
         style_key = f"pygment_{light_or_dark}_style"
-
-        # globalcontext sometimes doesn't exist so this ensures we do not error
         theme_name = _get_theme_options(app).get(style_key, None)
-        if theme_name is None and hasattr(app.builder, "globalcontext"):
-            theme_name = app.builder.globalcontext.get(f"theme_{style_key}")
-
+        if theme_name is None:
+            theme_name = app.builder.theme.get_options()[style_key]
```

Comment:
Note that you might be able to use this helper function we use elsewhere:

https://github.com/pydata/pydata-sphinx-theme/blob/58caad4710cfac7960d06aff7c88f034a2bea83a/src/pydata_sphinx_theme/__init__.py#L35

If calling `get_options()` is the "right" way to do this, maybe we should update that function too. Gah there are way too many ways to do the same thing in Sphinx

Reply:
I am already using that helper function 2 lines above; this line is only called if the style key wasn't found by the helper. Most of this you already know @choldgraf, but for completeness / future selves:

- [`app.builder.get_theme_config()`](https://github.com/sphinx-doc/sphinx/blob/036b69b9bb75a8294fd54c2e6f00f4106d5c450b/sphinx/builders/html/__init__.py#L298-L299
) returns
    - `app.builder.config.html_theme` (just the theme name)
    - `app.builder.config.html_theme_options` (dict of theme options). [initialized here](https://github.com/sphinx-doc/sphinx/blob/036b69b9bb75a8294fd54c2e6f00f4106d5c450b/sphinx/builders/html/__init__.py#L1330) as an empty dict, it's not clear to me when it's populated, and with what.

- `app.builder.theme_options` is a *copy* of `app.builder.config.html_theme_options`.
	- The copy is made [during `builder.init_templates()`](https://github.com/sphinx-doc/sphinx/blob/036b69b9bb75a8294fd54c2e6f00f4106d5c450b/sphinx/builders/html/__init__.py#L301-L305)
	- Apparently there are cases where that copy never happens (e.g., linkcheck?) so we have that helper function that checks `app.builder.config.html_theme_options` too.

- Calling `app.builder.theme.get_options()` will get just the values in `theme.conf` without any user overrides.
	- Normally it's called by sphinx as `.get_options(app.builder.theme_options)`, which makes me strongly suspect that `app.builder.theme_options` includes *only* the user-provided `html_theme_options` from their `conf.py` (and not the theme's defaults).
	- Here's where it's called during sphinx builds: [during `builder.prepare_writing()`](https://github.com/sphinx-doc/sphinx/blob/036b69b9bb75a8294fd54c2e6f00f4106d5c450b/sphinx/builders/html/__init__.py#L571-L574), right after `globalcontext` is created.
	- I suspect our problem has been that `builder.prepare_writing()` is sometimes never called (e.g. if no files have changed?). That would explain why we had to work around missing `globalcontext` and also why the theme's defaults weren't available/accessible.

In this case, I think it would be possible / sensible to update the helper function to also check the `theme.conf` values. Might be a good use case for `ChainMap`?

Reply:
Sphinx never ceases to amaze me lol. Thanks for putting all of this together!